packages/client/src/classes/client.js

```diff
@@ -86,47 +88,53 @@ class Client {
   /**
    * Create request
    */
-  createRequest(data) {
-
-    //Keep URL parameter consistent
-    if (data.uri) {
-      data.url = data.uri;
-      delete data.uri;
-    }
+  createRequest(opts) {
+
+    let options = {
+      url: opts.url || opts.uri,
+      baseURL: opts.baseURL || this.defaultRequest.baseURL,
+      method: opts.method || this.defaultRequest.method,
+      data: opts.data ? opts.data : opts.body ? opts.body : {},
+      params: opts.qs ? opts.qs : opts.params ? opts.params : {},
+      headers: opts.headers || {},
+    };
 
     //Merge data with empty request
-    const request = mergeData(this.defaultRequest, data);
+    options = mergeData(this.defaultRequest, options);
 
-    //Add headers
-    request.headers = this.createHeaders(request.headers);
-    return request;
+    options.headers = this.createHeaders(options.headers);
+    return options;
   }
 
   /**
    * Do a request
    */
-  request(data, cb) {
+  request(opts, cb) {
 
     //Create request
-    const request = this.createRequest(data);
+    opts = this.createRequest(opts);
+    this.lastResponse = undefined;
+    this.lastRequest = new Request(opts);
+    let _this = this;
 
     //Perform request
     const promise = new Promise((resolve, reject) => {
-      http(request, (error, response, body) => {
-
-        //Request error
-        if (error) {
+      axios(opts)
+        .then(response => {
+          // Successful response
+          _this.lastResponse = new Response(response.status, response.data, response.headers);
+          return resolve([_this.lastResponse, response.data]);
```

Comment:
This could result in threading issues if `lastResponse` get reassigned before returning.

Reply:
I think I fixed this problem, can you double check?